### ISSUE TITLE
CalibratedClassifierCV sigmoid calibration incorrect when base estimator is a Pipeline

### ISSUE THREAD

**Issue by huntermills707** [onnx/sklearn-onnx#1246]
## Description

When converting a `CalibratedClassifierCV` (with `method='sigmoid'`) to ONNX, the exported model produces **incorrect probability outputs** if the base estimator is a `sklearn.pipeline.Pipeline`. The predicted probabilities diverge significantly from sklearn's output (max absolute difference ~0.45, affecting 98%+ of samples).

The bug occurs with both `ensemble=True` and `ensemble=False`, and is triggered by **any** Pipeline wrapper — even one containing only an identity transform (`FunctionTransformer()`).

A bare classifier (not wrapped in a Pipeline) converts correctly.

## Root Cause (suspected)

The raw decision function scores are exported correctly. The sigmoid calibration parameters (`a`, `b`) appear to be correct. The issue is in how the ONNX graph applies the sigmoid transformation `expit(-(a * x + b))` when the calibrated estimator contains a Pipeline — it seems to apply the sigmoid to the wrong intermediate output or with incorrect parameters in the graph.

## Impact

This is a **silent** correctness bug — no error or warning is raised. The exported model produces plausible-looking probabilities that are significantly wrong.

## Workaround

Export the base estimator (not the calibrated wrapper) with `raw_scores=True`, then apply the sigmoid calibration manually at inference time:

```python
from scipy.special import expit

# Save a, b from the fitted calibrator
a = calibrator.calibrated_classifiers_[0].calibrators[0].a_
b = calibrator.calibrated_classifiers_[0].calibrators[0].b_

# Export inner estimator with raw scores
clf = calibrator.estimator
options = {id(clf): {"raw_scores": True}}
onx = convert_sklearn(clf, initial_types=initial_type, options=options)

# At inference time, apply sigmoid manually
raw_scores = onnx_session.run([label_name], {"X": X})
probabilities = expit(-(a * raw_scores + b))
```

## Minimal Reproducer

```python
import numpy as np
from sklearn.calibration import CalibratedClassifierCV
from sklearn.datasets import make_classification
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import FunctionTransformer
from skl2onnx import convert_sklearn
from skl2onnx.common.data_types import FloatTensorType
import onnxruntime as rt

X, y = make_classification(n_samples=2000, n_features=20, random_state=42)
X = np.abs(X).astype(np.float32)
X_train, X_test = X[:1500], X[1500:]
y_train, y_test = y[:1500], y[1500:]

clf = GradientBoostingClassifier(n_estimators=50, random_state=42)

estimators = {
    "bare classifier":        clf,
    "Pipeline(identity+clf)": Pipeline([("id", FunctionTransformer()), ("clf", clf)]),
}

for name, est in estimators.items():
    cal = CalibratedClassifierCV(est, method="sigmoid", ensemble=False)
    cal.fit(X_train, y_train)

    sk_proba = cal.predict_proba(X_test)[:, 1]

    onx = convert_sklearn(cal, initial_types=[("X", FloatTensorType([None, 20]))])
    sess = rt.InferenceSession(onx.SerializeToString())
    _, onnx_out = sess.run(None, {sess.get_inputs()[0].name: X_test})
    onnx_proba = onnx_out[:, 1] if isinstance(onnx_out, np.ndarray) else np.array([d[1] for d in onnx_out])

    diff = np.abs(sk_proba - onnx_proba)
    status = "OK" if diff.max() < 1e-4 else "BUG"
    print(f"[{status}] {name:30s}  max_diff={diff.max():.6f}  broken={(diff > 0.01).sum()}/{len(X_test)}")
```

### Output

```
[OK] bare classifier                 max_diff=0.000000  broken=0/500
[BUG] Pipeline(identity+clf)          max_diff=0.460038  broken=493/500
```

## Environment

- **skl2onnx**: 1.20.0
- **scikit-learn**: 1.8.0
- **onnxruntime**: 1.24.4
- **Python**: 3.12
